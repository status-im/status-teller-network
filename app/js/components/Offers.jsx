import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Card, CardHeader, CardBody } from 'reactstrap';
import { Link } from "react-router-dom";
import { withNamespaces } from 'react-i18next';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowRight, faEuroSign } from "@fortawesome/free-solid-svg-icons";

class Offers extends Component {
  renderOffers(t) {
    return this.props.offers.map((offer, index) => (
      <Card key={index} className="mb-2">
        <CardHeader>
          {offer.from}
          <FontAwesomeIcon icon={faArrowRight} className="mx-4"/>
          <FontAwesomeIcon icon={faEuroSign} className="mr-1"/>
          {offer.to}
        </CardHeader>
        <CardBody>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.type')}</dt>
              <dd>{offer.type}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.paymentMethods')}</dt>
              <dd>{offer.paymentMethod}</dd>
            </dl>
          </Row>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.location')}</dt>
              <dd>{offer.location}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.rate')}</dt>
              <dd>{offer.rate}</dd>
            </dl>
          </Row>
        </CardBody>
      </Card>
    ));
  }

  renderEmpty(t) {
    return (
      <Card body className="text-center">
        {t('offers.noOpen')}
      </Card>
    );
  }

  render() {
    const t = this.props.t;
    return (
      <Row className="mt-4">
        <Col xs="12">
          <span className="font-weight-bold h5">{t('offers.title')}</span>
          <Link to="/sell" className="float-right">{t('offers.create')}</Link>
        </Col>
        <Col xs="12">
          {this.props.offers.length === 0 ? this.renderEmpty(t) : this.renderOffers(t)}
        </Col>
      </Row>
    );
  }
}

Offers.propTypes = {
  t: PropTypes.func,
  offers: PropTypes.array
};

export default withNamespaces()(Offers);
