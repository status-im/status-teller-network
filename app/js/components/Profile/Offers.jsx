import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Card, CardHeader, CardBody } from 'reactstrap';
import { Link } from "react-router-dom";
import { withNamespaces } from 'react-i18next';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowRight, faEuroSign } from "@fortawesome/free-solid-svg-icons";

class Offers extends Component {
  renderOffers() {
    const {t, offers} = this.props;
    return offers.map((offer, index) => (
      <Card key={index} className="mb-2 shadow-sm">
        <CardHeader>
          {offer.token.symbol}
          <FontAwesomeIcon icon={faArrowRight} className="mx-4"/>
          <FontAwesomeIcon icon={faEuroSign} className="mr-1"/>
          {offer.currency}
        </CardHeader>
        <CardBody>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.type')}</dt>
              <dd>{t('offers.typeSell')}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.paymentMethods')}</dt>
              <dd>{offer.paymentMethodsForHuman}</dd>
            </dl>
          </Row>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.location')}</dt>
              <dd>{this.props.location}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.rate')}</dt>
              <dd>{offer.rateForHuman}</dd>
            </dl>
          </Row>
        </CardBody>
      </Card>
    ));
  }

  renderEmpty() {
    const {t} = this.props;
    return (
      <Card body className="text-center">
        {t('offers.noOpen')}
      </Card>
    );
  }

  render() {
    const {t, offers} = this.props;
    return (
      <div className="mt-3">
        <div>
          <h3 className="d-inline-block">{t('offers.title')}</h3>
          <span className="float-right">
            <Link to="/sell" className="float-right">{t('offers.create')} <FontAwesomeIcon icon={faArrowRight}/></Link>
          </span>
        </div>
        {offers.length === 0 ? this.renderEmpty() : this.renderOffers()}
      </div>
    );
  }
}

Offers.propTypes = {
  t: PropTypes.func,
  location: PropTypes.string,
  offers: PropTypes.array
};

export default withNamespaces()(Offers);
