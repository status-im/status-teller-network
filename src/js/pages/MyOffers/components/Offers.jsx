import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Card, CardHeader, CardBody, Button} from 'reactstrap';
import { Link } from "react-router-dom";
import { withNamespaces } from 'react-i18next';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowRight } from "@fortawesome/free-solid-svg-icons";
import ConfirmDialog from "../../../components/ConfirmDialog";
import {CURRENCY_DATA} from "../../../constants/currencies";
import {zeroAddress} from '../../../utils/address';
import NoArbitratorWarning from "../../../components/NoArbitratorWarning";
import classnames from 'classnames';
import iconDelete from '../../../../images/delete.svg';

class Offers extends Component {
  state = {
    displayDialog: false,
    offerId: null
  };

  confirmDelete = offerId => (e) => {
    if(e) e.preventDefault();
    this.setState({ offerId, displayDialog: true });
    return false;
  };

  displayDialog = show => () => {
    this.setState({displayDialog: show});
    return false;
  };


  deleteOffer = () => {
    this.props.deleteOffer(this.state.offerId);
    this.setState({offerId: null, displayDialog: false});
  };

  renderOffers(offers, enabled) {
    const {t} = this.props;
    return offers.map((offer, index) => (
      <Card key={index} className={classnames('mb-2', 'shadow-sm', {'card-transparent': !enabled})}>
        <CardHeader>
          {offer.token.symbol}
          <FontAwesomeIcon icon={faArrowRight} className="mx-4"/>
          <span className="text-small font-italic mr-2">{CURRENCY_DATA.find(x => x.id === offer.currency).symbol}</span>
          {offer.currency}
          { enabled && (
            <Button className="p-0 pl-3 pr-3 m-0 float-right btn-primary-outline" onClick={this.confirmDelete(offer.id)}>
              <img src={iconDelete} alt="Delete icon" />
            </Button>
          )}
        </CardHeader>
        <CardBody>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.type')}</dt>
              <dd>{t('offers.typeSell')}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.paymentMethods')}</dt>
              <dd>{offer.paymentMethodsForHuman}</dd>
            </dl>
          </Row>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.location')}</dt>
              <dd>{this.props.location}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.rate')}</dt>
              <dd>{offer.rateForHuman}</dd>
            </dl>
          </Row>
          <Row>
            <dl className="col-12">
              <dt>Arbitrator</dt>
              {offer.arbitrator === zeroAddress && <dd>
                <NoArbitratorWarning arbitrator={offer.arbitrator} label="Arbitrator not available. Offer will not appear in the offer list" />
              </dd>}
              {offer.arbitrator !== zeroAddress && <dd>{offer.arbitratorData.username} ({offer.arbitrator})</dd> }
            </dl>
          </Row>
        </CardBody>
      </Card>
    ));
  }

  renderEmpty() {
    const {t} = this.props;
    return (
      <Card body className="text-center">
        {t('offers.noOpen')}
      </Card>
    );
  }

  render() {
    const {t, offers} = this.props;
    const activeOffers = offers.filter(x => !x.deleted);
    const inactiveOffers = offers.filter(x => x.deleted);

    return (
      <div className="mt-3">
        <div>
          <h3 className="d-inline-block">{t('offers.active')}</h3>
          <span className="float-right">
            <Link to="/license" className="float-right text-small">{t('offers.create')}</Link>
          </span>
        </div>
        {activeOffers.length === 0 ? this.renderEmpty() : this.renderOffers(activeOffers, true)}
        <div className="mt-5">
          <h3 className="d-inline-block">{t('offers.past')}</h3>
        </div>
        {inactiveOffers.length === 0 ? this.renderEmpty() : this.renderOffers(inactiveOffers, false)}
        <ConfirmDialog display={this.state.displayDialog} onConfirm={this.deleteOffer} onCancel={this.displayDialog(false)} title="Delete offer" content="Are you sure?" cancelText="No" />
      </div>
    );
  }
}

Offers.propTypes = {
  t: PropTypes.func,
  location: PropTypes.string,
  offers: PropTypes.array,
  deleteOffer: PropTypes.func
};

export default withNamespaces()(Offers);
