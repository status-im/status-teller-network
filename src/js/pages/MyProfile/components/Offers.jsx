import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Row, Card, CardHeader, CardBody, Button} from 'reactstrap';
import { Link } from "react-router-dom";
import { withNamespaces } from 'react-i18next';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faArrowRight, faEllipsisV } from "@fortawesome/free-solid-svg-icons";
import ConfirmDialog from "../../../components/ConfirmDialog";
import {CURRENCY_DATA} from "../../../constants/currencies";


class Offers extends Component {
  state = {
    displayDialog: false,
    offerId: null
  }

  handleClose = () => {
    this.setState({ displayDialog: false });
  };

  confirmDelete = offerId => (e) => {
    if(e) e.preventDefault();
    this.setState({ offerId, displayDialog: true });
    return false;
  }

  displayDialog = show => () => {
    this.setState({displayDialog: show});
    return false;
  };


  deleteOffer = () => {
    this.props.deleteOffer(this.state.offerId);
    this.setState({offerId: null, displayDialog: false});
  }

  renderOffers() {
    const {t, offers} = this.props;
    return offers.map((offer, index) => (
      <Card key={index} className="mb-2 shadow-sm">
        <CardHeader>
          {offer.token.symbol}
          <FontAwesomeIcon icon={faArrowRight} className="mx-4"/>
          <span className="text-small font-italic mr-2">{CURRENCY_DATA.find(x => x.id === offer.currency).symbol}</span>
          {offer.currency}
          <Button className="p-0 pl-3 pr-3 m-0 float-right btn-link" onClick={this.confirmDelete(offer.id)}>
            <FontAwesomeIcon icon={faEllipsisV} />
          </Button>
        </CardHeader>
        <CardBody>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.type')}</dt>
              <dd>{t('offers.typeSell')}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.paymentMethods')}</dt>
              <dd>{offer.paymentMethodsForHuman}</dd>
            </dl>
          </Row>
          <Row>
            <dl className="col-6">
              <dt>{t('offers.location')}</dt>
              <dd>{this.props.location}</dd>
            </dl>
            <dl className="col-6">
              <dt>{t('offers.rate')}</dt>
              <dd>{offer.rateForHuman}</dd>
            </dl>
          </Row>
        </CardBody>
      </Card>
    ));
  }

  renderEmpty() {
    const {t} = this.props;
    return (
      <Card body className="text-center">
        {t('offers.noOpen')}
      </Card>
    );
  }

  render() {
    const {t, offers} = this.props;
    return (
      <div className="mt-3">
        <div>
          <h3 className="d-inline-block">{t('offers.title')}</h3>
          <span className="float-right">
            <Link to="/license" className="float-right">{t('offers.create')} <FontAwesomeIcon icon={faArrowRight}/></Link>
          </span>
        </div>
        {offers.length === 0 ? this.renderEmpty() : this.renderOffers()}
        <ConfirmDialog display={this.state.displayDialog} onConfirm={this.deleteOffer} onCancel={this.displayDialog(false)} title="Delete offer" content="Are you sure?" cancelText="No" />
      </div>
    );
  }
}

Offers.propTypes = {
  t: PropTypes.func,
  location: PropTypes.string,
  offers: PropTypes.array,
  deleteOffer: PropTypes.func
};

export default withNamespaces()(Offers);
