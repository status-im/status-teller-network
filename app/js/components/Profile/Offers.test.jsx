import React from "react";
import { shallow } from "enzyme";

import Offers from "./Offers";

describe('Offers', () => {
  it('should render correctly', () => {
    const component = shallow(<Offers location="London" offers={
      [
        {
          asset: '0x0',
          token: {
            symbol: 'ETH'
          },
          currency: 'EUR',
          paymentMethods: ['Credit Card'],
          margin: 1,
          marketType: 1
        }
      ]
    } />);
  
    expect(component).toMatchSnapshot();
  });

  it('should render when empty', () => {
    const component = shallow(<Offers offers={[]} />);
  
    expect(component).toMatchSnapshot();
  });
});
