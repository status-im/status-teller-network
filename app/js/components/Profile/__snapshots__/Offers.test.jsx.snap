// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Offers should render correctly 1`] = `
<div
  className="mt-3"
>
  <div>
    <h3
      className="d-inline-block"
    />
    <span
      className="float-right"
    >
      <Link
        className="float-right"
        replace={false}
        to="/sell"
      >
         
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f061",
                "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
              ],
              "iconName": "arrow-right",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </span>
  </div>
  <Card
    className="mb-2 shadow-sm"
    key="0"
    tag="div"
  >
    <CardHeader
      tag="div"
    >
      0x0
      <FontAwesomeIcon
        border={false}
        className="mx-4"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              448,
              512,
              Array [],
              "f061",
              "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
            ],
            "iconName": "arrow-right",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <FontAwesomeIcon
        border={false}
        className="mr-1"
        fixedWidth={false}
        flip={null}
        icon={
          Object {
            "icon": Array [
              320,
              512,
              Array [],
              "f153",
              "M310.706 413.765c-1.314-6.63-7.835-10.872-14.424-9.369-10.692 2.439-27.422 5.413-45.426 5.413-56.763 0-101.929-34.79-121.461-85.449h113.689a12 12 0 0 0 11.708-9.369l6.373-28.36c1.686-7.502-4.019-14.631-11.708-14.631H115.22c-1.21-14.328-1.414-28.287.137-42.245H261.95a12 12 0 0 0 11.723-9.434l6.512-29.755c1.638-7.484-4.061-14.566-11.723-14.566H130.184c20.633-44.991 62.69-75.03 117.619-75.03 14.486 0 28.564 2.25 37.851 4.145 6.216 1.268 12.347-2.498 14.002-8.623l11.991-44.368c1.822-6.741-2.465-13.616-9.326-14.917C290.217 34.912 270.71 32 249.635 32 152.451 32 74.03 92.252 45.075 176H12c-6.627 0-12 5.373-12 12v29.755c0 6.627 5.373 12 12 12h21.569c-1.009 13.607-1.181 29.287-.181 42.245H12c-6.627 0-12 5.373-12 12v28.36c0 6.627 5.373 12 12 12h30.114C67.139 414.692 145.264 480 249.635 480c26.301 0 48.562-4.544 61.101-7.788 6.167-1.595 10.027-7.708 8.788-13.957l-8.818-44.49z",
            ],
            "iconName": "euro-sign",
            "prefix": "fas",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      EUR
    </CardHeader>
    <CardBody
      tag="div"
    >
      <Row
        tag="div"
      >
        <dl
          className="col-6"
        >
          <dt />
          <dd />
        </dl>
        <dl
          className="col-6"
        >
          <dt />
          <dd />
        </dl>
      </Row>
      <Row
        tag="div"
      >
        <dl
          className="col-6"
        >
          <dt />
          <dd>
            London
          </dd>
        </dl>
        <dl
          className="col-6"
        >
          <dt />
          <dd />
        </dl>
      </Row>
    </CardBody>
  </Card>
</div>
`;

exports[`Offers should render when empty 1`] = `
<div
  className="mt-3"
>
  <div>
    <h3
      className="d-inline-block"
    />
    <span
      className="float-right"
    >
      <Link
        className="float-right"
        replace={false}
        to="/sell"
      >
         
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f061",
                "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
              ],
              "iconName": "arrow-right",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </span>
  </div>
  <Card
    body={true}
    className="text-center"
    tag="div"
  />
</div>
`;
